main.go

```diff
@@ -369,7 +369,9 @@ func main() {
 		r.Use(middleware.Recoverer)
 		r.Use(middleware.StripSlashes)
 		r.Use(middleware.Timeout(middlewareTimeout)) // best set per handler.
-		r.Use(middleware.Throttle(cfg.middleware.concurrentRequestLimit))
+		// With zero backlog concurrent requests crossing a rate-limit result in non-200 HTTP response.
+		backlogDuration := 1 * time.Millisecond
```

Comment:
Let's make this configurable. We might need to increase this duration without building a new binary.